overlay/lower/usr/share/udhcpc/default.script

```diff
@@ -74,7 +74,7 @@ case "$1" in
 			echo "nameserver $i" >>$tmpfile
 		done
 		mv $tmpfile $realconf
-		if [ -n "$ntpsrv" ] && [ -w /tmp/ntp.conf ]; then
+		if [ -n "$ntpsrv" ] && { [ ! -f /tmp/ntp.conf ] || [ -w /tmp/ntp.conf ]; }; then
```

Comment:
this won't work from root. any file will return true to `-w`.
```
[root@ing-wuuk-y0510-6a20 ~]# touch test; chmod 444 test; test -w test; echo $?
0
[root@ing-wuuk-y0510-6a20 ~]# ls -l
total 0
-r--r--r--    1 root     root             0 Aug 18 01:47 test
```
you need to check the actual permissions on the file
```
[root@ing-wuuk-y0510-6a20 ~]# stat -c%a test
444
```

Reply:
I was focused on the missing file scenario and missed that. Definitely wouldn't want to overwrite user preferences.    Thanks!